robot-server/robot_server/maintenance_runs/router/base_router.py

```diff
@@ -0,0 +1,270 @@
+    title: str = "No current run found"
+
+
+class RunAlreadyActive(ErrorDetails):
+    """An error if one tries to create a new run while one is already active."""
+
+    id: Literal["RunAlreadyActive"] = "RunAlreadyActive"
+    title: str = "Run Already Active"
+
+
+class ProtocolRunIsActive(ErrorDetails):
+    """An error if one tries to create a maintenance run while a protocol run is active."""
+
+    id: Literal["ProtocolRunIsActive"] = "ProtocolRunIsActive"
+    title: str = "Protocol Run Is Active"
+
+
+class RunNotIdle(ErrorDetails):
+    """An error if one tries to delete a run that is not idle."""
+
+    id: Literal["RunNotIdle"] = "RunNotIdle"
+    title: str = "Run is not idle."
+    detail: str = (
+        "Run is currently active. Allow the run to finish or"
+        " stop it with a `stop` action before attempting to modify it."
+    )
+
+
+class RunStopped(ErrorDetails):
+    """An error if one tries to modify a stopped run."""
+
+    id: Literal["RunStopped"] = "RunStopped"
+    title: str = "Run Stopped"
+
+
+class AllRunsLinks(BaseModel):
+    """Links returned along with a collection of runs."""
+
+    current: Optional[ResourceLink] = Field(
+        None,
+        description="Path to the currently active run, if a run is active.",
+    )
+
+
+async def get_run_data_from_url(
+    runId: str,
+    run_data_manager: MaintenanceRunDataManager = Depends(
+        get_maintenance_run_data_manager
+    ),
+) -> MaintenanceRun:
+    """Get the data of a maintenance run.
+
+    Args:
+        runId: Run ID pulled from URL.
+        run_data_manager: Current and historical run data management.
+    """
+    try:
+        run_data = run_data_manager.get(runId)
+    except RunNotFoundError as e:
+        raise RunNotFound(detail=str(e)).as_error(status.HTTP_404_NOT_FOUND)
+    except RunNotCurrentError as e:
+        raise RunStopped(detail=str(e)).as_error(status.HTTP_409_CONFLICT) from e
+
+    return run_data
+
+
+@base_router.post(
+    path="/maintenance_runs",
+    summary="Create a maintenance run",
+    description=dedent(
+        """
+        Create a new maintenance run to track robot interaction.
+
+        If a maintenance run already exists, it will be cleared
+        and a new one will be created.
+
+        Will raise an error if a protocol run exists and is not idle.
+        """
+    ),
+    status_code=status.HTTP_201_CREATED,
+    responses={
+        status.HTTP_201_CREATED: {"model": SimpleBody[MaintenanceRun]},
+        status.HTTP_409_CONFLICT: {"model": ErrorBody[ProtocolRunIsActive]},
+    },
+)
+async def create_run(
+    request_body: Optional[RequestModel[MaintenanceRunCreate]] = None,
+    run_data_manager: MaintenanceRunDataManager = Depends(
+        get_maintenance_run_data_manager
+    ),
+    run_id: str = Depends(get_unique_id),
+    created_at: datetime = Depends(get_current_time),
+    protocol_run_has_been_played: bool = Depends(get_protocol_run_has_been_played),
```

Comment:
the name is a bit confusing to me bc we dont play? should we change this to started? ```suggestion
    protocol_run_has_been_started: bool = Depends(get_protocol_run_has_been_started),
```

Reply:
This is for protocol run (not maintenance run), which can be 'play'ed and the information whether the protocol run has been played is important because creating a maintenance run is only allowed when protocol run has not received a play action yet.